api-client/src/test/java/com/xing/api/XingApiTest.java

```diff
@@ -57,6 +58,22 @@ public void getRightResourceImpl() throws Exception {
     }
 
     @Test
+    public void getResourceViaFactory() throws Exception {
+        XingApi api = new XingApi.Builder()
+              .custom()
+              .addResourceFactory(new Factory() {
+                  @Override public Resource create(Class<? extends Resource> cls, XingApi api) {
+                      return cls == FactoryResource.class ? new FactoryResource(api, "test") : null;
```

Comment:
`api.resource()` will never return `null` if it can't create a resource via a factory, it will fallback to the reflection api, if that fails it will throw.